docs/Design/Null Checking.md

```diff
@@ -0,0 +1,148 @@
+# Null Checking
+
+Creates checks for if the passed variable has value null. 

```

Comment:
What do you mean by "value null"? This should provide some sort of context + definition for what that means.